### PR TITLE
Change time for the nightly Circle CI workflow and update used packages [MAILPOET-6095]

### PR DESCRIPTION
**Pull request by lysyjan**
## Description

This PR slightly updates our Circle CI jobs.
- it changes the time of the nightly job
- it updates used versions of the 3rd plugins
- it adds using the oldest PHP and MySQL in the nightly job with premium

## Code review notes

_N/A_

## QA notes

We can skip QA here.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-6095]

## After-merge notes

_N/A_

## Tasks

- [x] 🚫 I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] 🚫 I added sufficient test coverage
- [x] 🚫 I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-6095]: https://mailpoet.atlassian.net/browse/MAILPOET-6095?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1060,3 +1060,32 @@ workflows:
...
+          codeception_image_version: 7.4-cli_20220605.0
+          wordpress_image_version: wp-6.4_php8.0_20231114.1
+          requires:
+            - build_premium
+      - unit_tests:
+          <<: *slack-fail-post-step
+          name: unit_with_premium_oldest
+          executor: wpcli_php_mysql_oldest
+          requires:
+            - build_premium
```
Comment (costasovo) [inline .circleci/config.yml]: Do we need this job? I mean, unit tests should test only one class or function with all other stuff mocked, so by this principle, they shouldn't be dependent on premium being active. I think that we don't even load WordPress for the unit tests.

**Review by costasovo — COMMENTED** [review]
@lysyjan Thanks for the update. The new schedule works better with the weekend gap because the results will always be from the same day. 👍 

I have a question about L-1 versions. I thought we wanted to automate the L-1 detection so that we don't have to hardcode and later manually update those. I see there is another ticket for detecting beta and RC versions but there is no for L-1. Shouldn't the automation of L-1 version download be the part of this ticket?

File: .circleci/config.yml (hunk 1 above)
Comment (lysyjan) [reply .circleci/config.yml]: Good point. I will update it.

**Comment by lysyjan** [comment]
> I have a question about L-1 versions. I thought we wanted to automate the L-1 detection so that we don't have to hardcode and later manually update those. I see there is another ticket for detecting beta and RC versions, but there is no ticket for L-1. Shouldn't the automation of L-1 version download be part of this ticket?

@costasovo We have [a different ticket](https://mailpoet.atlassian.net/browse/MAILPOET-6097) for automated updates, which contain WP, WC, and other plugins. It is in a separate ticket because it is a spike and should be time-boxed. We are not sure right now if it is possible to achieve our expectations. Does it make sense to you?

**Review by costasovo — APPROVED** [review]
@lysyjan Thanks for the explanation. I missed the ticket for research. It makes sense to do it timeboxed 👍

**Pull request merged**